dali/pipeline/data/tensor.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/pipeline/data/tensor.h"
+
+namespace dali {
+  /**
+ * Loads the Tensor with data from the input Tensor.
+ */
+template <typename Backend>
+template <typename InBackend>
+void Tensor<Backend>::Copy(const Tensor<InBackend> &other, AccessOrder order) {
+  constexpr bool is_host_to_host = std::is_same_v<Backend, CPUBackend> &&
+                                   std::is_same_v<InBackend, CPUBackend>;
+  auto src_order = other.order();
+  auto dst_order = order_;
+  if (!order) {
+    if (is_host_to_host)
+      order = AccessOrder::host();
+    else  // use device order, if available; if not, use whichever (dst, src) is set
+      order = dst_order.is_device()
+              ? dst_order
+              : src_order.is_device()
+                ? src_order
+                : dst_order ? dst_order : src_order;
+  }
```

Comment:
Maybe it's just me, but every time I see some fixes related to orders, I recall that similar, but slightly different patterns appear. Is there a generic way of handling that? Can we introduce a three-way merge function into order type (or as a free function)?

Something like: 
```
AccessOrder compute_work_order(AccessOrder dst_order, AccessOrder src_order, AccessOrder optional_work_order ={});
```
That would have good docstring explaining our intent here? I don't have a good idea how to convey host-host work there, but should that already be set on both dst_order and src_order?